<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mainWindow</class>
 <widget class="QMainWindow" name="mainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>971</width>
    <height>758</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Water Jug Puzzle</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>../../../.local/share/icons/McMuse-pink-dark/apps@2x/scalable/apps.com.qq.im.svg</normaloff>../../../.local/share/icons/McMuse-pink-dark/apps@2x/scalable/apps.com.qq.im.svg</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">* {
  background-color: rgb(98, 160, 234);
  color: rgb(255, 255, 255);
  font: 12pt &quot;Ubuntu Mono&quot;;
}

#sidebarWidget,
#sidebarWidget QLabel {
  background-color: rgb(26, 95, 180);
}

#statesTextBrowser {
  background-color: rgb(255, 255, 255);
  color: rgb(0, 0, 0);
}</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QWidget" name="sidebarWidget" native="true">
      <property name="minimumSize">
       <size>
        <width>260</width>
        <height>580</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>260</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0" alignment="Qt::AlignTop">
        <widget class="QLabel" name="descriptionLabel">
         <property name="autoFillBackground">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:600;&quot;&gt;Water Jug Puzzle&lt;/span&gt;&lt;/p&gt;&lt;p&gt;Water jug is a classic and also interesting puzzle to solve in free time with your friends. In this game, you are given some number of jugs with a capacity and a initial state of water in each jug. Now you should do some operations to make the water in one jug to be equal to the goal amount.&lt;br/&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt; font-weight:600;&quot;&gt;About Project&lt;/span&gt;&lt;/p&gt;&lt;p&gt;This project is one of the proposed programs for the artificial intelligence class that our professor said. Check out the repository on &lt;a href=&quot;https://github.com/sheikhartin/water-jug-puzzle-from-scratch&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;GitHub&lt;/span&gt;&lt;/a&gt; and enjoy the MIT license.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
         <property name="openExternalLinks">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="bodyWidget" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QFrame" name="setupFrame">
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="1" column="2">
           <widget class="QSpinBox" name="jugBAmountSpinBox">
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="value">
             <number>4</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QLabel" name="jugBLabel">
            <property name="text">
             <string>Jug B amount and capacity:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="goalLabel">
            <property name="text">
             <string>Goal amount to reach:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QSpinBox" name="jugACapacitySpinBox">
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="value">
             <number>14</number>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="jugALabel">
            <property name="text">
             <string>Jug A amount and capacity:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="2" colspan="2">
           <widget class="QSpinBox" name="goalAmountSpinBox">
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="value">
             <number>5</number>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QSpinBox" name="jugBCapacitySpinBox">
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="value">
             <number>6</number>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QSpinBox" name="jugAAmountSpinBox">
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="value">
             <number>11</number>
            </property>
           </widget>
          </item>
          <item row="5" column="3">
           <widget class="QPushButton" name="autoSolvePushButton">
            <property name="palette">
             <palette>
              <active>
               <colorrole role="WindowText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>255</red>
                  <green>255</green>
                  <blue>255</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="Button">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>98</red>
                  <green>160</green>
                  <blue>234</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="Text">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>255</red>
                  <green>255</green>
                  <blue>255</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="ButtonText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>255</red>
                  <green>255</green>
                  <blue>255</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="Base">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>98</red>
                  <green>160</green>
                  <blue>234</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="Window">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>98</red>
                  <green>160</green>
                  <blue>234</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="PlaceholderText">
                <brush brushstyle="NoBrush">
                 <color alpha="128">
                  <red>255</red>
                  <green>255</green>
                  <blue>255</blue>
                 </color>
                </brush>
               </colorrole>
              </active>
              <inactive>
               <colorrole role="WindowText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>255</red>
                  <green>255</green>
                  <blue>255</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="Button">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>98</red>
                  <green>160</green>
                  <blue>234</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="Text">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>255</red>
                  <green>255</green>
                  <blue>255</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="ButtonText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>255</red>
                  <green>255</green>
                  <blue>255</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="Base">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>98</red>
                  <green>160</green>
                  <blue>234</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="Window">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>98</red>
                  <green>160</green>
                  <blue>234</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="PlaceholderText">
                <brush brushstyle="NoBrush">
                 <color alpha="128">
                  <red>255</red>
                  <green>255</green>
                  <blue>255</blue>
                 </color>
                </brush>
               </colorrole>
              </inactive>
              <disabled>
               <colorrole role="WindowText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>255</red>
                  <green>255</green>
                  <blue>255</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="Button">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>98</red>
                  <green>160</green>
                  <blue>234</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="Text">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>255</red>
                  <green>255</green>
                  <blue>255</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="ButtonText">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>255</red>
                  <green>255</green>
                  <blue>255</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="Base">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>98</red>
                  <green>160</green>
                  <blue>234</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="Window">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>98</red>
                  <green>160</green>
                  <blue>234</blue>
                 </color>
                </brush>
               </colorrole>
               <colorrole role="PlaceholderText">
                <brush brushstyle="NoBrush">
                 <color alpha="128">
                  <red>255</red>
                  <green>255</green>
                  <blue>255</blue>
                 </color>
                </brush>
               </colorrole>
              </disabled>
             </palette>
            </property>
            <property name="toolTip">
             <string>Solves the puzzle automatically</string>
            </property>
            <property name="text">
             <string>Solve</string>
            </property>
           </widget>
          </item>
          <item row="5" column="2">
           <widget class="QComboBox" name="nextStateComboBox">
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="toolTip">
             <string>Choose the next state to apply</string>
            </property>
            <item>
             <property name="text">
              <string>-</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Empty both jugs</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Empty jug A</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Empty jug B</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Fill both jugs</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Fill jug A</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Fill jug B</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Pour jug A to B</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Pour jug B to A</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QTextBrowser" name="statesTextBrowser">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="html">
          <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Ubuntu Mono'; font-size:12pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="openExternalLinks">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="wishLabel">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt; font-style:italic;&quot;&gt;Wishing peace and goodness to all the waters of the world and the creatures that live in them...&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
